packages/react-native-sdk/src/components/Call/CallContent/CallContent.tsx

```diff
@@ -0,0 +1,123 @@
+import { StyleSheet, View } from 'react-native';
+import {
+  CallTopView as DefaultCallTopView,
+  CallTopViewProps,
+} from '../CallTopView';
+import {
+  CallParticipantsGrid,
+  CallParticipantsGridProps,
+  CallParticipantsSpotlight,
+} from '../CallLayout';
+import {
+  CallControls as DefaultCallControls,
+  CallControlProps,
+} from '../CallControls';
+import { CallParticipantsList as DefaultCallParticipantsList } from '../CallParticipantsList';
+import {
+  useCall,
+  useHasOngoingScreenShare,
+} from '@stream-io/video-react-bindings';
+import {
+  ParticipantNetworkQualityIndicator as DefaultParticipantNetworkQualityIndicator,
+  ParticipantReaction as DefaultParticipantReaction,
+  ParticipantLabel as DefaultParticipantLabel,
+  ParticipantVideoFallback as DefaultParticipantVideoFallback,
+  VideoRenderer as DefaultVideoRenderer,
+  ParticipantView as DefaultParticipantView,
+  LocalParticipantView as DefaultLocalParticipantView,
+} from '../../Participant';
+import { CallingState } from '@stream-io/video-client';
+import { useIncallManager } from '../../../hooks';
+
+export type CallContentProps = Pick<
+  CallTopViewProps,
+  'onBackPressed' | 'onParticipantInfoPress'
+> &
+  Pick<CallControlProps, 'onHangupCallHandler'> &
+  Pick<
+    CallParticipantsGridProps,
+    | 'CallParticipantsList'
+    | 'LocalParticipantView'
+    | 'ParticipantLabel'
+    | 'ParticipantNetworkQualityIndicator'
+    | 'ParticipantReaction'
+    | 'ParticipantVideoFallback'
+    | 'ParticipantView'
+    | 'VideoRenderer'
+  > & {
+    /**
+     * Component to customize the CallTopView component.
+     */
+    CallTopView?: React.ComponentType<CallTopViewProps>;
+    /**
+     * Component to customize the CallControls component.
+     */
+    CallControls?: React.ComponentType<CallControlProps>;
+  };
+
+export const CallContent = ({
+  onBackPressed,
+  onParticipantInfoPress,
+  onHangupCallHandler,
+  CallParticipantsList = DefaultCallParticipantsList,
+  LocalParticipantView = DefaultLocalParticipantView,
+  ParticipantLabel = DefaultParticipantLabel,
+  ParticipantNetworkQualityIndicator = DefaultParticipantNetworkQualityIndicator,
+  ParticipantReaction = DefaultParticipantReaction,
+  ParticipantVideoFallback = DefaultParticipantVideoFallback,
+  ParticipantView = DefaultParticipantView,
+  VideoRenderer = DefaultVideoRenderer,
+  CallTopView = DefaultCallTopView,
+  CallControls = DefaultCallControls,
+}: CallContentProps) => {
+  const hasScreenShare = useHasOngoingScreenShare();
+  /**
+   * This hook is used to handle IncallManager specs of the application.
+   */
+  useIncallManager({ media: 'video', auto: true });
+
+  const call = useCall();
+  const activeCallRef = useRef(call);
+  activeCallRef.current = call;
+
+  useEffect(() => {
+    return () => {
+      if (activeCallRef.current?.state.callingState !== CallingState.LEFT) {
+        activeCallRef.current?.leave();
+      }
+    };
+  }, []);
+
+  const participantViewProps = {
```

Comment:
```suggestion
  const participantViewProps: ParticipantProps = {
```

add some typing here so that future maintenance is easy

Reply:
In that case, we will have to create a separate type for all these component props. Are you fine with it?

Reply:
for the shared ones, yes!